File: Python/Searching/BinarySearch/LeetCode744.py
class Solution(object):
    def nextGreatestLetter(self, letters, target):
        left = 0
        right = len(letters) - 1
    
        while(left <= right):
            mid = (left + right)//2
            if(letters[mid] == target):
                left = mid + 1
            
            elif(letters[mid] < target):
                left = mid + 1
            else:
                right = mid -1
            
        if(left == len(letters)):
            return letters[0]
        
        return letters[left]
    

if __name__ == '__main__':
    letters = ['c','f','j'];
    target = 'c';
    
    sol = Solution()
    
    print(sol.nextGreatestLetter(letters, target))
    

File: Javascript/Searching/BinarySearch/src/LeetCode744.ts
function nextGreatestLetter(letters: string[], target: string): string {
    let left = 0;
    let right = letters.length - 1;


    while(left <= right){
        let mid = Math.floor((left + right)/2);

        if(letters[mid] == target){
            left = mid + 1;
        }

        else if(letters[mid] > target){
            right = mid -1;
        }

        else{
            left = mid + 1;
        }
    }
    
    if(left == letters.length){
        return letters[0];
    }

    return letters[left]; 
};

let letters = ['c','f','j'];
let target = 'c';
console.log(nextGreatestLetter(letters, target));